src/pages/SingleCakePage.js

```diff
@@ -0,0 +1,51 @@
+import "./SingleCakePage.css";
+import Header from "../components/Header";
+import Footer from "../components/Footer";
+import ItemCard from "../components/ItemCard";
+
+export default function SingleCakePage() {
+  return (
+    <div className="Single-page">
+      <Header />
```

Comment:
If the header is in every page, you should move it oustide the `Switch` and outside this component. 
This way, you only rerender the parts that change.
This would mean your pages would need to care about the header or the footer, they would just care about the content.

Reply:
Very good point. Changed it as per suggested.